### PR TITLE
add event-model and event_type-mode

### PR DESCRIPTION
**Pull request by eedy8888**

### CODE REVIEW

**Comment by chez-shanpu** [comment]
Models should be defined in `models.py`
Like this: https://docs.djangoproject.com/ja/3.1/intro/overview/#design-your-model

**Review by chez-shanpu — CHANGES_REQUESTED** [review]
https://github.com/naist-sushi/sushi-library/pull/34#issuecomment-773817369

And rebase or merge latest master branch

**Comment by eedy8888** [comment]
I fix it, please check it, thanks.

**Comment by eedy8888** [comment]
@chez-shanpu  I fix it, please check it, thanks.

**Review by chez-shanpu — CHANGES_REQUESTED** [review]
@eedy8888 please resolve conflicts

**Comment by eedy8888** [comment]
@chez-shanpu  please check it again, thanks.

File: sushi_library/library/models.py [hunk 1]
```diff
@@ -56,3 +57,75 @@ class Meta:
                              on_delete=models.CASCADE)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+
+class event(models.Model):
+    """Model representing a event ."""
+    title = models.CharField(max_length=200)
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, help_text='Unique ID for this particular book across whole library')
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:66]: No need.

File: sushi_library/library/models.py [hunk 2]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+    eventtype = models.CharField(
+        max_length=10,
+        choices=EVENT_STATUS,
+        blank=True,
+        default='r',
+        help_text='Book availability',
+    )
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:73]: Use `models.ForeignKey`

File: sushi_library/library/models.py [hunk 3]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:74]: Add foreign-key of `Book` and `User`, and `created_at`

File: sushi_library/library/models.py [hunk 4]
```diff
@@ -56,3 +57,75 @@ class Meta:
                              on_delete=models.CASCADE)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+
+class event(models.Model):
+    """Model representing a event ."""
+    title = models.CharField(max_length=200)
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:65]: What is title?

File: sushi_library/library/models.py [hunk 5]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+    )
+    
+    
+EVENT_STATUS = (
+        ('r', 'requested'),
+        ('ar', 'approve the requesting'),
+        ('rr', 'requesting rejected'),
+        ('b', 'borrowed by someone'),
+        ('s', 'stored to lab')
+    )
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:82]: No need.

File: sushi_library/library/models.py [hunk 6]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+   
+    
+   
+
+   
+
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:88]: Too many empty lines.

File: sushi_library/library/models.py [hunk 7]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+
+
+class event_type (models.Model):
+    """Model representing an event."""
+    
+    bookname = models.CharField(max_length=100)
+    imprint = models.CharField(max_length=200)
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, help_text='Unique ID for this particular book across whole library')
+    requested_date = models.DateField(null=True, blank=True)
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:101]: No need.

File: sushi_library/library/models.py [hunk 8]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+    
+    eventtype = models.CharField(
+        max_length=10,
+        choices=EVENT_STATUS,
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:105]: No need.

File: sushi_library/library/models.py [hunk 9]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+class event_type (models.Model):
+    """Model representing an event."""
+    
+    bookname = models.CharField(max_length=100)
+    imprint = models.CharField(max_length=200)
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, help_text='Unique ID for this particular book across whole library')
+    requested_date = models.DateField(null=True, blank=True)
+    
+    eventtype = models.CharField(
+        max_length=10,
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:104]: `max_length=255`

File: sushi_library/library/models.py [hunk 10]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+        choices=EVENT_STATUS,
+        blank=True,
+        default='r',
+        help_text='Book availability',
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:108]: No need.

File: sushi_library/library/models.py [hunk 11]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+    )
+    
+    
+EVENT_STATUS = (
+        ('r', 'requested'),
+        ('ar', 'approve the requesting'),
+        ('rr', 'requesting rejected'),
+        ('b', 'borrowed by someone'),
+        ('s', 'stored to lab')
+    )
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:118]: No need.

File: sushi_library/library/models.py [hunk 12]
```diff
@@ -56,3 +57,75 @@ class Meta:
                              on_delete=models.CASCADE)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+
+class event(models.Model):
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:63]: Add Meta class and set `db_table`

File: sushi_library/library/models.py [hunk 13]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+        ('ar', 'approve the requesting'),
+        ('rr', 'requesting rejected'),
+        ('b', 'borrowed by someone'),
+        ('s', 'stored to lab')
+    )
+
+
+
+    class Meta:
+        ordering = ['requested_date']
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:123]: Put Meta class on the next line after `class event_type (models.Model):`.
`ordering` is not needed.
`db_table` is needed.

File: sushi_library/library/models.py [hunk 14]
```diff
@@ -56,3 +57,75 @@ class Meta:
...
+
+   
+
```
Comment (chez-shanpu) [inline sushi_library/library/models.py:126]: Too many lines.

**Review by chez-shanpu — CHANGES_REQUESTED** [review]
@eedy8888 
Please check my review.
And next time, please request a review after the 'check' has passed.
Thanks.

**Comment by chez-shanpu** [comment]
Working on it in #38.
So, I'll close this.

**Pull request closed without merging**